libs/core/src/lib/micro-process-flow/components/micro-process-flow/micro-process-flow.component.ts

```diff
@@ -0,0 +1,246 @@
+    @Input()
+    transitionSpeed = 0.3;
+
+    /** Pagination transition effect */
+    @Input()
+    transitionTimingFunction = 'ease';
+
+    /** Micro process flow items. */
+    @ContentChildren(MicroProcessFlowItemComponent)
+    items: QueryList<MicroProcessFlowItemComponent>;
+
+    /** Previous items outside the viewport. */
+    _previousItemsCount = 0;
+
+    /** Next items outside the viewport. */
+    _nextItemsCount = 0;
+
+    /** Should show next button. */
+    get _showNextButton(): boolean {
+        return this._nextItemsCount > 0;
+    }
+
+    /** Should show previous button. */
+    get _showPreviousButton(): boolean {
+        return this._previousItemsCount > 0;
+    }
+
+    /** @hidden */
+    @ViewChild('wrapperContainer')
+    private _wrapperContainer: ElementRef<HTMLElement>;
+
+    /** @hidden */
+    @ViewChild('container')
+    private _container: ElementRef<HTMLElement>;
+
+    /** @hidden */
+    @ViewChild('goNextButton')
+    private _goNextButton: ElementRef<HTMLElement>;
+
+    /** @hidden */
+    private _currentOffset = 0;
+
+    /** @hidden */
+    private _isRtl = false;
+
+    /** @hidden */
+    private get _paginationDirection(): number {
+        return this._isRtl ? 1 : -1;
+    }
+
+    /** @hidden */
+    private _subscriptions = new Subscription();
+
+    /** @hidden */
+    constructor(
+        private _cd: ChangeDetectorRef,
+        @Optional() private _rtl: RtlService,
+        @Optional() private _contentDensityService: ContentDensityService
+    ) { }
+
+    /** @hidden */
+    ngOnInit(): void {
+        if (this._rtl) {
+            this._subscriptions.add(this._rtl.rtl.subscribe((value) => {
+                this._isRtl = value;
+
+                if (this._showPreviousButton) {
+                    this._paginate(0);
+                }
+            }));
+        }
+
+        if (this.compact === undefined && this._contentDensityService) {
+            this._subscriptions.add(this._contentDensityService._contentDensityListener
+                .subscribe(density => {
+                    this.compact = density !== 'cozy';
+                    this._paginate(0);
+                })
+            );
+        }
+    }
+
+    /** @hidden */
+    ngAfterContentInit(): void {
+        this.listenOnItemsChange();
+    }
+
+    /** @hidden */
+    ngOnDestroy(): void {
+        this._subscriptions.unsubscribe();
+    }
+
+    /** Listens on items change and checks if navigation buttons should be visible. */
+    listenOnItemsChange(): void {
+        this._subscriptions.add(this.items.changes.pipe(startWith(0)).subscribe(() => {
+            // Fixes expression changed error.
+            setTimeout(() => {
+                this.items.forEach((item) => {
+                    item.setFinalStep(false);
+                });
+
+                this.items.last?.setFinalStep(true);
+                this._setNavigationButtons();
+                this._paginate(0);
+            });
```

Comment:
You can fix it without setTimeout. Invoke `listenOnItemsChange` inside another hook - `ngAfterViewInit`. I did it and remove setTimeout, works well.